vignettes/projecting_incidence.Rmd

```diff
@@ -0,0 +1,292 @@
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, 
+                      message = FALSE, 
+                      warning = FALSE, 
+                      collapse = TRUE,
+                      comment = "#>",
+                      dpi = 300
+                      )
+
+```
+
+## Overview
+
+Branching processes can be used to project infectious disease trends provided 
+we can characterize the distribution of times between 
+successive cases (serial interval), and the distribution of secondary cases 
+produced by a single individual (offspring distribution). Such simulations can 
+be achieved in `bpmodels` with the `chain_sim()` function. @Pearson2020, and 
+@abbott2020 illustrate its application to COVID-19. 
+
+The purpose of this vignette is to use early data on COVID-19 in South Africa 
+[@marivate2020] to illustrate how `bpmodels` can be used to forecast an 
+outbreak. 
+
+
+Let's load the required packages
+
+```{r packages, include=TRUE}
+library("bpmodels")
+library("dplyr")
+library("ggplot2")
+library("lubridate")
+```
+
+## Data
+
+We will get and clean the first $15$ days of the COVID-19 
+outbreak in South Africa to seed the simulation for this example.
+
+```{r data, message=FALSE}
+data_url <- "https://raw.githubusercontent.com/dsfsi/covid19za/master/data/covid19za_timeline_confirmed.csv" # nolint: line_length_linter. 
+
+#Read the data in using the url
+covid19_sa <- read.csv(data_url)
+
+# Subset the first 15 days and count the number of cases per date
+covid19_sa <- covid19_sa %>% 
+  select(date) %>% 
+  mutate(date = lubridate::dmy(date)) %>%
+  filter(date <= min(date) + lubridate::days(15)) %>%   
+  group_by(date) %>% 
+  summarise(cases = n()) %>%   
+  ungroup()
+```
+
+## Inputs  
+Using the data above, we will set up a vector of start times for each case.
+
+```{r linelist_gen, message=FALSE}
+days_since_index <- as.integer(covid19_sa$date - min(covid19_sa$date))
+
+start_times <- unlist(mapply(
+  function(x, y) rep(x, times = ifelse(y == 0, 1, y)),
+  days_since_index,
+  covid19_sa$cases
+))
+                       
+```
+
+
+Additionally, `chain_sim()` requires the end time for the simulations and the 
+maximum size of each chain. Since each result of `chain_sim()` is stochastic,
+it is also best to run it many times. 
+
+We will specify these as follows:
+
+```{r input_prep2, message=FALSE}
+#' Date to end simulation (14 day projection in this case)
+
+projection_window <- 14 # 14 days/ 2-week ahead projection
+
+projection_end_day <- max(days_since_index) + projection_window
+
+#' Number of simulations
+sim_rep <- 100
+
+#' Maximum chain size allowed
+chain_threshold <- 1000
+
+```
+
+### Serial interval
+
+In this example, we will assume based on COVID-19 literature that the 
+serial interval, $\mathcal{S}$, is log-normal distributed with parameters, 
+$\mu = 4.7$ and $\sigma = 2.9$ [@Pearson2020]. The log-normal mean, 
+$E[ \mathcal{S} ]$ and standard deviation $SD[ \mathcal{S} ]$ are 
```

Comment:
The `\mathcal{S}` didn't render for me locally. Perhaps something to look into if it's dependent on different character encoding settings.

Reply:
It seems to be a MacOS thing. I'll revise the parts using \mathcal.